src/main/java/games/strategy/thread/ThreadPool.java

```diff
@@ -1,59 +1,69 @@
 package games.strategy.thread;
 
 import java.util.ArrayDeque;
+import java.util.Queue;
+import java.util.concurrent.ExecutionException;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 import java.util.concurrent.Future;
 
-import games.strategy.util.ThreadUtil;
+import games.strategy.debug.ClientLogger;
 
 /**
  * An ExecutorService backed thread pool.
  */
 public class ThreadPool {
   private final ExecutorService executorService;
-  private final ArrayDeque<Future<?>> futuresStack = new ArrayDeque<>();
+  private final Queue<Future<?>> futureQueue = new ArrayDeque<>();
 
   /**
-   * Creates a new instance of ThreadPool max is the maximum number of threads the pool can have. The pool may have
-   * fewer threads at any given time.
+   * Creates a thread pool that reuses a fixed number of threads
+   * operating off a shared unbounded queue. At any point, at most
+   * {@code nThreads} threads will be active processing tasks.
+   * If additional tasks are submitted when all threads are active,
+   * they will wait in the queue until a thread is available.
+   * If any thread terminates due to a failure during execution
+   * prior to shutdown, a new one will take its place if needed to
+   * execute subsequent tasks. The threads in the pool will exist
+   * until it is explicitly {@link ThreadPool#shutDown shutdown}.
+   *
+   * @param nThreads the number of threads in the pool
```

Comment:
I believe you meant `max` instead of `nThreads` here, above, and below (or you meant to change the name of the actual parameter).

Reply:
That's what happens when I try to copy-paste javadocs ^^
Good catch